### PR TITLE
Rewrite ModalService (EventEmitter -> useState)

### PR DESCRIPTION
**Pull request by gasiortomasz**

### CODE REVIEW

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 1]
```diff
@@ -2,14 +2,16 @@ import React from 'react';
...
 
 interface OnboardingProps {
   sdk: UniversalLoginSDK;
   onConnectClick: (...args: any[]) => void;
   onCreateClick: (...args: any[]) => void;
+  modalService: ModalService;
 }
 
-export const Onboarding = ({sdk, onConnectClick, onCreateClick}: OnboardingProps) => {
+export const Onboarding = ({sdk, onConnectClick, onCreateClick, modalService}: OnboardingProps) => {
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]: `const modalService = useModal();`

File: universal-login-wallet/src/ui/react/CreateAccount/CreateAccount.tsx [hunk 2]
```diff
@@ -5,13 +5,17 @@ import {useServices, useWalletConfig} from '../../hooks';
 import {WalletSelector} from '@universal-login/react';
 import {WalletSuggestionAction, defaultDeployOptions} from '@universal-login/commons';
 import Modal from '../Modals/Modal';
+import {useModal} from '../../hooks/useModal';
+import ModalService from '../../../core/entities/ModalService';
 
 interface CreateAccountProps {
   location?: {state: {from: {pathname: string}}};
+  modalService: ModalService;
```
Comment (JustynaBroniszewska) [inline universal-login-wallet/src/ui/react/CreateAccount/CreateAccount.tsx]: remove

File: universal-login-wallet/src/ui/hooks/useModal.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import {useState} from 'react';
+import ModalService, {ModalStateType} from '../../core/entities/ModalService';
+
+export const useModal = (): ModalService => {
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/hooks/useModal.ts]: Extract to commons and add generic type? #nextPR

File: universal-login-react/src/ui/Modals/Modal.tsx [hunk 4]
```diff
@@ -2,13 +2,14 @@ import React from 'react';
...
-const Modal = () => {
-  const {modalService} = useServices();
-  const openModal = useSubscription(modalService);
-  switch (openModal) {
+export interface ModalProps {
+  modalService: ModalService;
+}
+
+const Modal = ({modalService}: ModalProps) => {
+  switch (modalService.modalState) {
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Modals/Modal.tsx]: use `useModal` instead of passing as an arg

**Pull request merged**